### PR TITLE
feat(select): add header actions slot

### PR DESCRIPTION
**Pull request by pedrorebelocosta**

### CODE REVIEW

File: packages/ui-components/src/components/multi-select-dropdown/multi-select-dropdown.scss [hunk 1]
```diff
@@ -23,4 +23,16 @@
...
+
+	.selection-clear {
+		@include kv-font-span-regular;
+		user-select: none;
+		color: kv-color('neutral-2');
+		cursor: pointer;
+	
+		&.disabled {
+			color: kv-color('neutral-5');
+			cursor: auto;
```
Comment (djcouto) [inline packages/ui-components/src/components/multi-select-dropdown/multi-select-dropdown.scss:35]: When the clear selection button is disabled, don't we want to show a disallowed cursor?

File: packages/ui-components/src/components/single-select-dropdown/single-select-dropdown.scss [hunk 2]
```diff
@@ -23,4 +23,16 @@
...
+
+	.selection-clear {
+		@include kv-font-span-regular;
+		user-select: none;
+		color: kv-color('neutral-2');
+		cursor: pointer;
+	
+		&.disabled {
+			color: kv-color('neutral-5');
+			cursor: auto;
```
Comment (djcouto) [inline packages/ui-components/src/components/single-select-dropdown/single-select-dropdown.scss:35]: Same as before, shouldn't it be 'not-allowed'?

File: packages/ui-components/src/components/select/select.tsx [hunk 3]
```diff
@@ -17,64 +16,41 @@ export class KvSelect implements ISelect, ISelectEvents {
...
 	/** @inheritdoc */
-	@Prop({ reflect: true }) selectionClearEnabled?: boolean;
-	/** @inheritdoc */
-	@Prop({ reflect: true }) clearSelectionLabel?: string = SELECT_CLEAR_SELECTION_LABEL;
-	/** @inheritdoc */
 	@Prop({ reflect: true }) minHeight?: string;
 	/** @inheritdoc */
 	@Prop({ reflect: true }) maxHeight?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) hasHeaderActions?: boolean = false;
```
Comment (djcouto) [inline packages/ui-components/src/components/select/select.tsx:25]: Do we really need this? Isn't this just the presence of a slot?

```ts
private hasActionsSlot = () => {
		return !isNil(this.el.querySelector('[slot="select-header-actions"]'));
}
```

File: packages/ui-components/src/components/select/select.tsx [hunk 4]
```diff
@@ -17,64 +16,41 @@ export class KvSelect implements ISelect, ISelectEvents {
...
 		this.searchChange.emit(event.detail);
 	};
 
-	private onClearSelection = () => {
-		this.clearSelection.emit();
-	};
-
-	private getCustomStyle = () =>
-		omitBy(
+	private get customStyle() {
```
Comment (djcouto) [inline packages/ui-components/src/components/select/select.tsx:34]: Any particular reason to change this to a get? Or is it just a syntax preference?

**Review by djcouto — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/select/select.tsx (hunk 4 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/select/select.tsx:34]: just syntax preference, can revert if you prefer :D

File: packages/ui-components/src/components/single-select-dropdown/single-select-dropdown.scss (hunk 2 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/single-select-dropdown/single-select-dropdown.scss:35]: I'll check with the design team :)

File: packages/ui-components/src/components/select/select.tsx [hunk 5]
```diff
@@ -17,64 +16,41 @@ export class KvSelect implements ISelect, ISelectEvents {
...
-									onClick={this.onClearSelection}
-								>
-									{this.clearSelectionLabel}
-								</div>
-							)}
-						</div>
-					)}
+					<div class={{ 'select-header-container': this.searchable || this.hasHeaderActions }}>
+						{this.searchable && <kv-search value={this.searchValue} placeholder={this.searchPlaceholder} onTextChange={this.onSearchChange} />}
+						{this.hasHeaderActions && <slot name="select-header-actions" />}
```
[the remark this answers is not in the document]
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/select/select.tsx:50]: Thanks for the feedback, I'll be making some changes to the component! :)

File: packages/ui-components/src/components/multi-select-dropdown/multi-select-dropdown.scss (hunk 1 above)
Comment (pedrorebelocosta) [reply packages/ui-components/src/components/multi-select-dropdown/multi-select-dropdown.scss]: Waiting for confirmation on this from design.

Comment (pedrorebelocosta) [reply packages/ui-components/src/components/multi-select-dropdown/multi-select-dropdown.scss]: Confirmed :)

**Review by djcouto — APPROVED** [review]

**Review by GoncaloCamaz — APPROVED** [review]

**Review by miguelpinto98 — APPROVED** [review]

**Pull request merged**